File: 667B.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int a, b, x, y, n;
        ll ans;
        cin >> a >> b >> x >> y >> n;
        if (b > a)
            swap(a, b);
        if (b - y >= n)
        {
            b -= n;
            ans = a * b;
            cout << ans << '\n';
        }
        else
        {
            n -= (b - y);
            b = y;
            a = max(a - n, x);
            ans = (ll)a * (ll)b;
            cout << ans << '\n';
        }
    }
}

File: 667B.py
t = int(input())
for _ in range(t):
    a, b, x, y, n = map(int, input().split())
    aa = a
    bb = b
    nn = n
    if(a-x >= n):
        a -= n
        ans = a * b
    else:
        n -= (a-x)
        a = x
        b = max(b - n, y)
        ans = a * b
    if(bb-y >= nn):
        bb -= nn
        ans = min(ans, aa * bb)
    else:
        nn -= (bb - y)
        bb = y
        aa = max(aa - nn, x)
        ans = min(ans, aa * bb)
    print(ans)
